docs/src/design/proposals/Registration.md

```diff
@@ -0,0 +1,518 @@
+   to impose minimal constraints. For instance, it should not be
+   responsible for performing script-loading. Its sole responsibility
+   is to facilitate communication between components.
+5. The Registration API should continue to support ubiquitous use of
+   patterns that have proven useful for plugin-based extensibility
+   (the Registry, Composite, and Decorator patterns, specifically.)
+   However, it should be designed such that knowledge of these patterns
+   is only required when it is appropriate to the specific task or
+   activity at hand. (For instance, you should not need to be familiar
+   with decorators in order to simply register something.)
+
+# Interfaces
+
+In keeping with the scope of the Registration API, the interfaces
+described here are sufficient to:
+
+* Describe the set of plugins in use for a particular instance of
+  Open MCT, and initiate their behavior.
+* Support common patterns by which plugins can utilize and expose
+  defined extension points.
+
+Notably, there is no interdependency between these two sets of
+behavior; one could use the base classes for extension points
+independently of the plugin mechanism, and vice versa. This both
+ensures loose coupling within the Registration API, and also
+allows for greater flexibility for developers implementing plugins.
+
+## Application-level Interfaces
+
+```nomnoml
+[Application |
+    install(plugin : Plugin)
+    uninstall(plugin : Plugin)
+    run()
+]
+
+[Plugin |
+    initialize()
+    start()
+]
+
+[Application]<:-[MCT |
+    core : Plugin
+    ui : Plugin
+    policy : Plugin
+    ...etc
+]
+[Application]-o[Plugin]
+```
+
+Summary of interfaces:
+
+* `Application` represents a complete piece of software that has been
+  composed of plugins.
+  * `install(plugin)` adds a plugin to this application.
+  * `uninstall(plugin)` removes a plugin from this application.
+  * `run()` starts the application. This will first initialize all
+    plugins, then start all plugins.
+* `Plugin` represents a unit of functionality available for use within
+  applications. It exposes methods to be triggered at various points
+  in the application lifecycle. A plugin is meant to be "single-use";
+  multiple calls to `initialize()` and/or `start()` should have no
+  effect.
+  * `initialize()` performs any configuration and/or registration
+    associated with this plugin.
+  * `start()` initiates any behavior associated with this plugin that
+    needs to run immediately after the application has started. (Useful
+    for a "bootstrap" plugin.)
+* `MCT` extends `Application` and self-installs various
+  plugins during its constructor call. It also exposes these same
+  plugins as public fields such that other applications may access
+  them (to `uninstall` them, for instance, or to pass them into other
+  plugins.)
+
+Rationale for various interfaces:
+
+* `Application` separates out a core responsibility of Open MCT (plugin
```

Comment:
`Application` might be an unnecessary abstraction, I'm not sure. Are we really building a generic application framework?